osu.Game/Skinning/LegacySkin.cs

```diff
@@ -60,10 +68,12 @@ private class LegacySkinResourceStore : IResourceStore<byte[]>
 
             private string getPathForFile(string filename)
             {
+                bool hasExtension = filename.Contains('.');
+
                 string lastPiece = filename.Split('/').Last();
 
                 var file = skin.Files.FirstOrDefault(f =>
-                    string.Equals(Path.GetFileNameWithoutExtension(f.Filename), lastPiece, StringComparison.InvariantCultureIgnoreCase));
+                    string.Equals(hasExtension ? f.Filename : Path.GetFileNameWithoutExtension(f.Filename), lastPiece, StringComparison.InvariantCultureIgnoreCase));

```

Comment:
What's this doing?

Reply:
If the incoming lookup has an extension then it respects that for lookups, else it drops it during matching.

Reply:
But hasExtension implies you have a '.'. If you don't have a '.' then you... get the file without the extension? But there should already be no extension since there's no '.', so at least one of these calls is redundant.

Reply:
it's running that on the lookup value, not the input parameter

Reply:
:brain: 💥 :into_space:

Reply:
ok fair enough

Reply:
Filenames are stored with extensions in the database storage, which we want to ignore here if the query value *doesn't* have an extension.